File: backend/src/main/java/de/snickit/rezeptebackend/entities/Category.java
package de.snickit.rezeptebackend.entities;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "category", schema = "public")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long categoryId;

    @ManyToMany
    @JoinTable(
            name = "recipe_categories",
            joinColumns = @JoinColumn(name = "category_id"),
            inverseJoinColumns = @JoinColumn(name = "recipe_id"))
    private Set<Recipe> recipes = new HashSet<>();
}

File: backend/src/main/kotlin/de/snickit/recipe/backend/persistence/entities/Category.kt
package de.snickit.recipe.backend.persistence.entities

import java.util.*
import javax.persistence.*

@Entity
data class Category (
        var categoryName: String? = null
) {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val categoryId: Long? = null

    @ManyToMany
    @JoinTable(name = "recipe_categories",
            joinColumns = [JoinColumn(name = "category_id")],
            inverseJoinColumns = [JoinColumn(name = "recipe_id")],
            foreignKey = ForeignKey(name = "category_reccat_fk"),
    inverseForeignKey = ForeignKey(name = "reccat_category_fk"))
    private var recipes: Set<Recipe> = HashSet()
}
